sled-hardware/src/disk.rs

```diff
@@ -148,12 +149,14 @@ impl UnparsedDisk {
         slot: i64,
         variant: DiskVariant,
         identity: DiskIdentity,
+        is_boot_disk: bool,
```

Comment:
I'm not sure this is the best way to relay this information. Another option I considered is change `DiskVariant` (i.e., `DiskVariant::M2 { is_boot_disk: bool }`). Not sure which would be nicer to use from sled-agent though.

Reply:
I think it's reasonable to have it in the same structure as the "slot" - the sled agent will be selecting for "all M.2s" before querying "which one is the boot disk" anyway, so I don't think it'll make much of a difference.